### PR TITLE
Image Extraction Scalability Improvements

### PR DESCRIPTION
**Pull request by kevinsoucy**
*Issue #49 , if available:*

*Description of changes:*

1. The name lengths are exceeding 64 chars for iam roles, due to poorly defined manifests, the name:

    addf-ros-image-demo-ford-analysis-ford-pipeline-dag-ap-southeast-1
    will now be:
    addf-ros-image-demo-dags-ford-dag-ap-southeast-1
    by changing the group name to "dags" and the module name to "ford"

2. Refactor Image Extraction to use EBS Volumes for task storage and EC2 capacity providers for ECS, as AWS Batch does not support increasing ephemeralStorage for fargate tasks, and with large files the tasks run out of disk space.

3. Image upload task uses ThreadPools for uploading files in parallel, increasing upload speeds by 10x or more.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: manifests/ros-image-demo/deployment.yaml [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
+  - name: dags
+    path: manifests/ros-image-demo/image-pipeline-modules.yaml
 targetAccountMappings:
   - alias: primary
     accountId: 1234567890
     default: true
     parametersGlobal:
       dockerCredentialsSecret: aws-addf-docker-credentials
     regionMappings:
-      - region: us-west-2
```
Comment (dgraeber) [inline manifests/ros-image-demo/deployment.yaml:21]: what will be the impact of CI by this change (if any)?

**Comment by dgraeber** [comment]
The validation is failing due to not having a check if the launch_template_name exists

Other than that, LGTM

**Comment by kevinsoucy** [comment]
> The validation is failing due to not having a check if the launch_template_name exists
> 
> Other than that, LGTM

Hey @dgraeber  - all good from my side as well. I tested the batch module and pipeline with and without the ebs volumes and think this ready to merge.

**Review by dgraeber — APPROVED** [review]

File: modules/analysis/rosbag-image-pipeline/app.py [hunk 2]
```diff
@@ -33,43 +33,42 @@ def _param(name: str) -> str:
...
-if not full_access_policy:
-    raise ValueError("S3 Full Access Policy ARN is missing.")
-
-if not on_demand_job_queue and not spot_job_queue and not fargate_job_queue:
-    raise ValueError("Requires at least one job queue.")
+image_topics = os.getenv(_param("IMAGE_TOPICS"), "{}")
+sensor_topics = os.getenv(_param("SENSOR_TOPICS"), "{}")
+
+#
+# if not png_batch_job_def_arn:
```
Comment (srinivasreddych) [inline modules/analysis/rosbag-image-pipeline/app.py:43]: Can you remove the commented code if is no longer needed?

File: modules/analysis/rosbag-image-pipeline/stack.py [hunk 3]
```diff
@@ -107,13 +107,13 @@ def __init__(
...
                 ],
             ),
-            iam.PolicyStatement(
-                actions=[
-                    "iam:PassRole",
-                ],
-                effect=iam.Effect.ALLOW,
-                resources=[object_detection_role, lane_detection_role],
-            ),
+            # iam.PolicyStatement(
```
Comment (srinivasreddych) [inline modules/analysis/rosbag-image-pipeline/stack.py:110]: Can you remove the commented code if is no longer needed?

File: modules/sensor-extraction/ros-to-png/src/main.py [hunk 4]
```diff
@@ -42,16 +45,16 @@
...
-    def __init__(self, topic, images_path):
-        self.bridge = CvBridge()
-        output_dir = os.path.join(images_path, topic.replace("/", "_"))
-        self.video = f'/tmp/{topic.replace("/", "_")}/video.mp4'
-        logger.info("Get video for topic {}".format(topic))
-        logger.info(
-            f"""ffmpeg -r 20 -f image2 -i {output_dir}/frame_%06d.png \
-            -vcodec libx264 -crf 25  -pix_fmt yuv420p {self.video}"""
-        )
+# class VideoFromBag:
```
Comment (srinivasreddych) [inline modules/sensor-extraction/ros-to-png/src/main.py:48]: Can you remove the commented code if is no longer needed?

File: modules/sensor-extraction/ros-to-png/stack.py [hunk 5]
```diff
@@ -52,9 +52,6 @@ def __init__(
             **kwargs,
         )
 
-        if platform not in ["FARGATE", "EC2"]:
```
Comment (srinivasreddych) [inline modules/sensor-extraction/ros-to-png/stack.py:55]: Is is not good to have this validation in place?

**Comment by srinivasreddych** [comment]
Hey @kevinsoucy Pls look into the above comments, other than that it looks good to me

File: manifests/ros-image-demo/deployment.yaml (hunk 1 above)
Comment (kevinsoucy) [reply manifests/ros-image-demo/deployment.yaml]: reverted

File: modules/analysis/rosbag-image-pipeline/app.py (hunk 2 above)
Comment (kevinsoucy) [reply modules/analysis/rosbag-image-pipeline/app.py]: Good catch, I uncommented this again, I needed to comment it out to destroy one of my existing deployments.

File: modules/sensor-extraction/ros-to-png/src/main.py (hunk 4 above)
Comment (kevinsoucy) [reply modules/sensor-extraction/ros-to-png/src/main.py:48]: uncommented

**Review by srinivasreddych — APPROVED** [review]

**Comment by kevinsoucy** [comment]
Thanks @dgraeber , @srinivasreddych , could one of you please merge this?

**Pull request merged**